apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -543,13 +470,76 @@ contract DisputableVoting is DisputableApp {
     }
 
     /**
-    * @dev Void an entry
+    * @dev Void a vote
     * @param _voteId Identification number of the entry to be voided
     */
     function _onDisputableActionVoided(uint256 _voteId) internal {
+        // When a vote that has been challenged gets voided, it is considered allowed for the current implementation.
+        // This could be the case for challenges where the arbitrator refuses to rule.
         _onDisputableActionAllowed(_voteId);
     }
 
+    // Internal fns
+
+    /**
+    * @dev Internal function to create a new vote
+    * @return voteId id for newly created vote
+    */
+    function _newVote(bytes _executionScript, string _metadata) internal returns (uint256 voteId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1; // avoid double voting in this very block
+        uint256 votingPower = token.totalSupplyAt(snapshotBlock);
+        require(votingPower > 0, ERROR_NO_VOTING_POWER);
+
+        voteId = votesLength++;
+
+        Vote storage vote_ = votes[voteId];
+        vote_.startDate = getTimestamp64();
+        vote_.snapshotBlock = snapshotBlock;
+        vote_.overruleWindow = overruleWindow;
+        vote_.supportRequiredPct = supportRequiredPct;
+        vote_.minAcceptQuorumPct = minAcceptQuorumPct;
+        vote_.votingPower = votingPower;
+        vote_.executionScript = _executionScript;
+        vote_.disputableStatus = DisputableStatus.Active;
+
+        // Notify the Agreement app tied to the current voting app about the vote created.
+        // This is mandatory to make votes disputable, we then store its reference on the Agreement app.
+        vote_.actionId = _newAgreementAction(voteId, msg.sender, bytes(_metadata));
+
+        emit StartVote(voteId, msg.sender, _metadata);
+    }
+
+    /**
+    * @dev Internal function to cast a vote. It assumes the queried vote exists.
+    */
+    function _vote(uint256 _voteId, bool _supports, address _voter) internal {
+        // This could re-enter, though we can assume the governance token is not malicious
+        Vote storage vote_ = votes[_voteId];
+        _castVote(vote_, _voteId, _supports, _voter);
+        _overwriteCasterIfNecessary(vote_, _voter);
+    }
+
+    /**
+    * @dev Internal function for a representative to cast a vote on behalf of many voters. It assumes the queried vote exists.
+    */
+    function _voteOnBehalfOf(address[] _voters, uint256 _voteId, bool _supports) internal {
```

Comment:
Still kind of find it weird that this interface isn't

```
_voteOnBehalfOf(uint256 _voteId, bool _supports, address[] _voters)
```

To align it with the previous one :).